iceoryx_binding_c/include/iceoryx_binding_c/log.h

```diff
@@ -29,6 +29,14 @@ enum iox_LogLevel
     Iceoryx_LogLevel_Fatal
 };
 
+#ifdef _WIN32
+#define Iceoryx_LogLevel_Verbose Iceoryx_LogLevel_Trace
+#else
+#define Iceoryx_LogLevel_Verbose                                                                                       \
+    _Pragma("GCC warning \"'Iceoryx_LogLevel_Verbose' is deprecated in 3.0 and will be removed in 4.0. Please use "    \
+            "'Iceoryx_LogLevel_Trace'\"") Iceoryx_LogLevel_Trace
```

Comment:
Compiling this with cyclonedds gives the following error:

```
/src/cyclonedds/src/core/ddsi/src/ddsi_shm_transport.c: In function ‘to_iox_loglevel’:
/src/cyclonedds/src/core/ddsi/src/ddsi_shm_transport.c:59:40: error: _Pragma takes a parenthesized string literal
   59 |         case DDSI_SHM_VERBOSE : return Iceoryx_LogLevel_Verbose;
      |                                        ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from src/cyclonedds/src/core/ddsi/src/ddsi_shm_transport.c:13:
/src/core/ddsi/src/ddsi_shm_transport.c:59:40: error: ‘_Pragma’ undeclared (first use in this function)
   59 |         case DDSI_SHM_VERBOSE : return Iceoryx_LogLevel_Verbose;
      |                                        ^~~~~~~~~~~~~~~~~~~~~~~~
/src/cyclonedds/src/core/ddsi/src/ddsi_shm_transport.c:59:40: note: each undeclared identifier is reported only once for each function it appears in
/src/cyclonedds/src/core/ddsi/src/ddsi_shm_transport.c:59:40: error: expected ‘;’ before ‘)’ token
   59 |         case DDSI_SHM_VERBOSE : return Iceoryx_LogLevel_Verbose;
      |                                        ^~~~~~~~~~~~~~~~~~~~~~~~
src/cyclonedds/src/core/ddsi/src/ddsi_shm_transport.c:59:40: error: expected statement before ‘)’ token
gmake[2]: *** [src/core/CMakeFiles/ddsc.dir/build.make:1210: src/core/CMakeFiles/ddsc.dir/ddsi/src/ddsi_shm_transport.c.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:689: src/core/CMakeFiles/ddsc.dir/all] Error 2
gmake: *** [Makefile:156: all] Error 2
```

I guess the handling with newline and the pragma is a bit tricky.

Shorter version that works:

```suggestion
#define Iceoryx_LogLevel_Verbose                                                                                \
    _Pragma("GCC warning \"Iceoryx_LogLevel_Verbose' is deprecated and will be removed in the next release.\"") \
        Iceoryx_LogLevel_Trace
```

Reply:
@dkroenke should be fine now